lib/chef/http/authenticator.rb

```diff
@@ -167,46 +168,79 @@ def self.get_cert_password
         @win32registry = Chef::Win32::Registry.new
         path = "HKEY_LOCAL_MACHINE\\Software\\Progress\\Authentication"
         # does the registry key even exist?
-        present = @win32registry.get_values(path)
-        if present.nil? || present.empty?
+        # password_blob should be an array of hashes
+        password_blob = @win32registry.get_values(path)
+        if password_blob.nil? || password_blob.empty?
           raise Chef::Exceptions::Win32RegKeyMissing
         end
 
-        present.each do |secret|
-          if secret[:name] == "PfxPass"
-            password = decrypt_pfx_pass(secret[:data])
-            return password
+        # Did someone have just the password stored in the registry?
+        raw_data = password_blob.map { |x| x[:data] }
+        vector = raw_data[2]
+        if !!vector
+          decrypted_password = decrypt_pfx_pass_with_vector(password_blob)
+        else
+          decrypted_password = decrypt_pfx_pass_with_password(password_blob)
+          if !!decrypted_password
+            migrate_pass_to_use_vector
+          else
```

Comment:
Why not just re-encrypt the password the right way?

```suggestion
          decrypted_password = decrypt_pfx_pass_with_password(password_blob)
          if !!decrypted_password
            migrate_pass_to_use_vector(decyrpted_password)
          else
```

Reply:
That's what this does.

Reply:
As we chatted about in the meeting, I don't think so. We don't pass in the existing password in, so it can't be encrypting the password using the new method. Instead it makes a new random password.